src/librustdoc/doctest/make.rs

```diff
@@ -445,40 +444,33 @@ fn parse_source(
     parent_dcx: Option<DiagCtxtHandle<'_>>,
     span: Span,
 ) -> Result<ParseSourceInfo, ()> {
-    use rustc_errors::DiagCtxt;
-    use rustc_errors::emitter::{Emitter, HumanEmitter};
-    use rustc_span::source_map::FilePathMapping;
-
     let mut info =
         ParseSourceInfo { already_has_extern_crate: crate_name.is_none(), ..Default::default() };
 
     let wrapped_source = format!("{DOCTEST_CODE_WRAPPER}{source}\n}}");
 
-    let filename = FileName::anon_source_code(&wrapped_source);
-
-    let sm = Arc::new(SourceMap::new(FilePathMapping::empty()));
-    let translator = rustc_driver::default_translator();
-    info.supports_color =
-        HumanEmitter::new(stderr_destination(ColorConfig::Auto), translator.clone())
-            .supports_color();
-    // Any errors in parsing should also appear when the doctest is compiled for real, so just
-    // send all the errors that the parser emits directly into a `Sink` instead of stderr.
-    let emitter = HumanEmitter::new(AutoStream::never(Box::new(io::sink())), translator);
```

Comment:
It made no sense to use a `HumanEmitter` here that sends things into the void. I'm pretty sure that leads to extra work (rendering the diagnostic?) that gets thrown away immediately after. Use a `SilentEmitter` instead which exists precisely for such use cases.